app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/service/notification/AppTpDisabledContentPlugin.kt

```diff
@@ -0,0 +1,91 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.mobile.android.vpn.service.notification
+
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import androidx.core.app.NotificationCompat
+import androidx.core.app.TaskStackBuilder
+import com.duckduckgo.di.scopes.AppScope
+import com.duckduckgo.mobile.android.vpn.AppTpVpnFeature
+import com.duckduckgo.mobile.android.vpn.R
+import com.duckduckgo.mobile.android.vpn.VpnFeaturesRegistry
+import com.duckduckgo.mobile.android.vpn.service.TrackerBlockingVpnService
+import com.duckduckgo.mobile.android.vpn.service.VpnStaticNotificationContentPlugin
+import com.duckduckgo.mobile.android.vpn.service.VpnStaticNotificationContentPlugin.NotificationContent
+import com.duckduckgo.mobile.android.vpn.service.VpnStaticNotificationContentPlugin.NotificationPriority
+import com.duckduckgo.mobile.android.vpn.service.VpnStaticNotificationContentPlugin.NotificationPriority.NORMAL
+import com.duckduckgo.mobile.android.vpn.service.VpnStaticNotificationContentPlugin.Type
+import com.duckduckgo.mobile.android.vpn.service.VpnStaticNotificationContentPlugin.Type.DISABLED
+import com.duckduckgo.mobile.android.vpn.service.VpnReminderReceiver
+import com.duckduckgo.mobile.android.vpn.ui.notification.NotificationActionReportIssue
+import com.duckduckgo.mobile.android.vpn.ui.tracker_activity.DeviceShieldTrackerActivity
+import com.squareup.anvil.annotations.ContributesMultibinding
+import javax.inject.Inject
+
+@ContributesMultibinding(AppScope::class)
+class AppTpDisabledContentPlugin @Inject constructor(
+    private val context: Context,
+    private val vpnFeaturesRegistry: VpnFeaturesRegistry,
+) : VpnStaticNotificationContentPlugin {
+    private val notificationPendingIntent by lazy { getAppTPNotificationPressIntent(context) }
+    private val notificationActionPendingIntent by lazy { getAppTPStartIntent(context) }
+
+    override fun getContent(): NotificationContent? {
+        return NotificationContent(
+            isSilent = true,
+            shouldAutoCancel = null,
+            customViewLayout = R.layout.notification_device_shield_disabled,
+            onNotificationPressIntent = notificationPendingIntent,
+            notificationAction = listOf(
+                NotificationCompat.Action(
+                    R.drawable.ic_vpn_notification_24,
+                    context.getString(R.string.atp_EnableCTA),
+                    notificationActionPendingIntent,
+                ),
+                NotificationActionReportIssue.reportIssueNotificationAction(context),
+            ),
+        )
+    }
+
+    override fun getPriority(): NotificationPriority {
+        return NORMAL
+    }
+
+    override fun isActive(): Boolean {
+        return vpnFeaturesRegistry.getLastRegistryChange().isAppTpDisabled()
```

Comment:
Ditto, why not `vpnFeaturesRegistry.registryChanges().last()`.
We need to do a `runBlocking` probably, but at least is explicit

Reply:
When I tried this before, the flow doesn't emit any values. I can check again tomorrow - maybe I missed something then.